### DISCUSSION TITLE
ZSH not working anymore after upgrading to 5.9

### DISCUSSION THREAD

**Discussion by alexanderharm** [Homebrew/discussions#3290] [Everyday usage] [answered]
Today I upgraded my ZSH to 5.9 via Homebrew (I use ZSH with Antigen). However, the upgrade seems to have messed up my installation completely. I have to add that I work as a non-admin user.

So when launching Terminal I am immediately confronted with error messages like:
- compinit fails: `compinit:480: compdump: function definition file not found`
- compaudit fails as well: `compaudit: No directories in $fpath, cannot continue`
- even zsh-newuser-install fails: `zsh: command not found: zsh-newuser-install`

I have no idea what is going on... Anyone any ideas?

**Comment by alexanderharm** [comment]
Deleted my `.zshrc` and ran `zsh-newuser-install` again.

Error is still the same: `compinit:480: compdump: function definition file not found`

**Comment by alexanderharm** [comment]
I guess the issue is this:

>compinit: A "y" response to the "Ignore ... and continue?" prompt removes insecure elements from the set of completion functions, where previously it ignored the compaudit result and included all elements.

Found in the release notes here <https://zsh.sourceforge.io/releases.html>

**Comment by alexanderharm** [comment, marked as answer]
For anyone encountering the same issue, I don't really have a fix combining compaudit and Homebrew. Since the errors were driving me nuts I simply disabled the error check.

You can do this with `compinit -C`.

Or in Antigen set an environment variable `ANTIGEN_COMPINIT_OPTS="-C"`

**Reply by beansh** [reply]
Thanks for that link to the release notes!

Perhaps a better option for this scenario would be `compinit -u`.  This option avoids the security checks and makes use of all files found under $fpath, regardless of ownership. In this case, the files are owned by a separate admin brew user who is not root.

The issue with option -C is if the dumpfile already exists it wont be updated, even if the are new or modified completions after that initial creation. To update, the dumpfile would first have to be deleted then regenerated.

[The Z Shell Manual -> 20 Completion System -> 20.2.1 Use of compinit](https://zsh.sourceforge.io/Doc/Release/Completion-System.html)
